src/material-experimental/mdc-select/select.scss

```diff
@@ -116,3 +118,25 @@ $mat-select-placeholder-arrow-space: 2 * ($mat-select-arrow-size + $mat-select-a
     display: block;
   }
 }
+
+// Our MDC form field implementation is based on the MDC text field which doesn't include styles
+// for select. The select specific styles are not present as they don't use their text field as a
+// container. Below are the styles to account for the select arrow icon at the end.
+.mat-mdc-form-field-type-mat-select.mat-form-field-appearance-fill {
+  .mdc-floating-label {
+    max-width: calc(100% - #{$mat-select-placeholder-arrow-space});
+  }
+  .mdc-floating-label--float-above {
+    max-width: calc(100% / #{$scale} - #{$mat-select-placeholder-arrow-space} / #{$scale});
```

Comment:
nit: move everything besides the 100% into a compile time expression:
`calc(100% / #{$scale - $mat-select-placeholder-arrow-space / $scale})`